core/src/main/scala/cats/ContravariantMonoidal.scala

```diff
@@ -0,0 +1,32 @@
+package cats
+
+import simulacrum.typeclass
+
+/**
+ * [[ContravariantMonoidal]] functors are functors that supply
+ * a unit along the diagonal map for the `contramap2` operation.
+ *
+ * Must obey the laws defined in cats.laws.ContravariantMonoidalLaws.
+ *
+ * Based on ekmett's contravariant library:
+ * https://hackage.haskell.org/package/contravariant-1.4/docs/Data-Functor-Contravariant-Divisible.html
+ */
+@typeclass trait ContravariantMonoidal[F[_]] extends ContravariantSemigroupal[F] { self =>
+  /**
+   * `unit` produces an instance of `F` for any type `A`
+   * that is trivial with respect to `contramap2` along
+   * the diagonal
+   */
+  def unit[A]: F[A]
+
+  def liftContravariant[A, B](f: A => B): F[B] => F[A] =
+    ContravariantMonoidal.contramap2(unit[B], _: F[B])(((b: B) => (b, b)) compose f)(self, self)
+
+  // Technically, this is not correct, as the Applicative is composed with the ContravariantMonoidal, not the other way around
```

Comment:
Would you mind elaborate a bit more (or point to a more detailed discussion/explanation) here?

Reply:
Sure, I can also just change the name and move this possibly. The point here is mostly that I didn't want to add anything to `Applicative` about `ContravariantMonoidal` since this might live in `cats.more`, so I ended up having this composition live here in `ContravariantMonoidal` even though the composition is `Applicative compose ContravariantMonoidal`, so the name and the signature are a little out of sync with what I saw in the rest of the repository.

Reply:
ah, I propose we rename and move it to `Applicative`

Reply:
Got it, will do. Thanks.